### DISCUSSION TITLE
Can't access from external IP

### DISCUSSION THREAD

**Discussion by onosendi** [fastify/fastify#4401] [Q&A] [answered]
Express:

```js
const express = require('express');

const app = express();

app.get('/', (req, res) => {
	res.send('Foo');
});

app.listen(8001, () => {
	console.log('Listening on 8001');
});
```

Fastify:

```js
const fastify = require('fastify')({ logger: true });

fastify.get('/', async (request, reply) => {
  return { hello: 'world' }
});

fastify.listen({ port: 8001 });
```

I get a response from Express when I `http <ip>:8001/`, but not for Fastify.

**Comment by onosendi** [comment, marked as answer]
Seems `fastify.listen({ port: 8001, host: '0.0.0.0' })` fixes it.

**Reply by Uzlopak** [reply]
0.0.0.0 means to listen on all possible ip adresses. 
So probably you should check if you can reach the server under another IP. btw. What do you get if you do `fastify.listen({ port:8001}, (err, address) => console.log(address))`?

**Reply by onosendi** [reply]
@Uzlopak http://127.0.0.1:8001

**Comment by jsumners** [comment]
https://www.fastify.io/docs/latest/Reference/Server/#listen

**Reply by onosendi** [reply]
I read this thoroughly before I posted here. Where specifically does this help me? Thanks.

**Reply by jsumners** [reply]
https://github.com/fastify/fastify/blob/de53fba42a664af05f1492564f5a71427135a55b/docs/Reference/Server.md?plain=1#L907-L928
